sql/src/main/java/io/crate/operation/collect/sources/InformationSchemaIterables.java

```diff
@@ -81,7 +83,14 @@ public InformationSchemaIterables(final Schemas schemas,
         partitionInfos = new PartitionInfos(clusterService);
         columnsIterable = tablesIterable.transformAndConcat(ColumnsIterable::new);
 
-        constraints = tablesIterable.filter(i -> i != null && i.primaryKey().size() > 0);
+        FluentIterable<ConstraintInfo> primaryKeyConstraints = tablesIterable
+            .filter(i -> i != null && i.primaryKey().size() > 0)
+            .transform(t -> new ConstraintInfo(
+                t.ident(),
+                String.format(Locale.ENGLISH, "%s_pk", t.ident().name()),
```

Comment:
Could be replaced with `t.ident().name() + "_pk"` to avoid the `String.format` overhead.